src/java.base/share/classes/java/lang/reflect/InvocationHandler.java

```diff
@@ -54,41 +54,44 @@ public interface InvocationHandler {
      * @param   proxy the proxy instance that the method was invoked on
      *
      * @param   method the {@code Method} instance corresponding to
-     * the interface method invoked on the proxy instance.  The declaring
-     * class of the {@code Method} object will be the interface that
-     * the method was declared in, which may be a superinterface of the
-     * proxy interface that the proxy class inherits the method through.
+     * the method invoked on the proxy instance.  The invoked method may be
+     * one of the {@link Object#hashCode() hashCode}, {@link
+     * Object#equals(Object) equals}, and {@link Object#toString() toString}
+     * methods declared in {@link Object}, or any public non-static method
+     * declared in one of the proxy interfaces or a superinterface of them.
+     * See also the {@linkplain Proxy##duplicate-methods "Methods Duplicated in
+     * Multiple Proxy Interfaces"} section in {@code Proxy}.
```

Comment:
The proposed wording looks okay for when a method is invoked on a proxy instance. 

InvocationHandler is not strictly tied to Proxy. I'm just wondering if the we should qualify this so that the text is in the context invoking a method on the proxy object. I'm also wondering if it should be moved to the interface description to avoid multi sentence parameter description, e.g. "When invoked on a proxy object, the method ...".  Up to you but I suspect there are wider usages of this interface.

Reply:
Indeed, I think cglib uses InvocationHandler for their subclass proxies. Since you suggested to move this to the interface, I have started reviewing Proxy, and found some other intriguing properties that I don't know if it best belongs to this effort here. Will describe in another post.